tests/pom.xml

```diff
@@ -422,6 +453,63 @@ Test cases are run against Docker images of Hono server + (Apache Qpid Dispatch
               </execution>
             </executions>
           </plugin>
+          <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>build-helper-maven-plugin</artifactId>
+            <version>3.0.0</version>
+            <executions>
+              <execution>
+                <id>reserve-network-port</id>
+                <goals>
+                  <goal>reserve-network-port</goal>
+                </goals>
+                <phase>initialize</phase>
+                <configuration>
+                  <randomPort>true</randomPort>
+                  <portNames>
+                    <!--
+                      The port on which Kafka's advertised listener for the Docker-external clients is set up
+                      must be the same on which Docker exposes the container port to the "outside".
+                      The docker-maven-plugin cannot be used since the advertised listener is defined as an
+                      environment variable of the container and the plugin chooses the random port after the definition
+                      of the container is made.
+                    -->
+                    <portName>kafka.port</portName>
+                  </portNames>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
+          <plugin>
+            <!--
+              The standard Maven plugin for writing properties cannot be used as it is only capable of writing
+              all project properties to file but not specific ones only.
```

Comment:
The other components use the docker-maven-plugin for writing dynamically assigned ports to property files. But for Kafka we cannot rely on the docker-maven-plugin to pick a port because of how Kafka's advertised listener works (see comment on the build-helper-maven-plugin and its reserve-network-port goal)

Reply:
I see